### PR TITLE
Models

### LINKED ISSUE #3: Entity : Comment

### LINKED ISSUE #4: Entity : ToDont

### LINKED ISSUE #5: Entity : User

### LINKED ISSUE #30: Relation : ToDont participant

### PR DESCRIPTION
**Pull request by g-wizzy**
Close #3 
Close #4 
Close #5 
Close #30

### CODE REVIEW

File: src/main/java/ch/hearc/todont/models/PledgeKey.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((toDontId == null) ? 0 : toDontId.hashCode());
+        result = prime * result + (int) (userId ^ (userId >>> 32));
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
```
Comment (Blatoy) [inline src/main/java/ch/hearc/todont/models/PledgeKey.java]: Should we always use {} even for one-line ifs? I'll check if we can have a linter performs checks in our pipeline

File: src/main/java/ch/hearc/todont/models/Pledge.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import javax.persistence.MapsId;
+
+import org.springframework.lang.Nullable;
+
+/**
+ * This contains the additional information in the relation
+ * between a user and a ToDont he's pledged to follow
+ */
+@Entity
+public class Pledge {
```
Comment (Blatoy) [inline src/main/java/ch/hearc/todont/models/Pledge.java:19]: Do we need a `@Table(name = "Pledge ")` here?

File: src/main/java/ch/hearc/todont/models/PledgeKey.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package ch.hearc.todont.models;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+
+@Embeddable
+public class PledgeKey implements Serializable{
```
Comment (Blatoy) [inline src/main/java/ch/hearc/todont/models/PledgeKey.java:10]: Maybe missing table name here?

File: src/main/java/ch/hearc/todont/models/User.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    @Column(
+        name = "email",
+        nullable = false
+    )
+    private String email;
+
+    @Column(
+        name = "password",
+        length = 255,
```
Comment (Blatoy) [inline src/main/java/ch/hearc/todont/models/User.java:39]: Depending on the hash we will use, I'm not sure we need a 255 characters long field here

**Review by Blatoy — CHANGES_REQUESTED** [review]

File: src/main/java/ch/hearc/todont/models/PledgeKey.java (hunk 3 above)
Comment (g-wizzy) [reply src/main/java/ch/hearc/todont/models/PledgeKey.java:10]: This is a key type, and so not a table in itself.

File: src/main/java/ch/hearc/todont/models/User.java (hunk 4 above)
Comment (g-wizzy) [reply src/main/java/ch/hearc/todont/models/User.java:39]: I left it knowing we might change it later on. As long as we don't start deploying before changing this length to an appropriate value, there's no problem. We just have to not forget it.

**Review by Blatoy — APPROVED** [review]

**Review by TitoOn — APPROVED** [review]

**Pull request merged**